### ISSUE TITLE
Support for committing images

### ISSUE THREAD

**Issue by AnimeAllstar** [Himenon/github-api-create-commit#8]
I've been trying to commit an uploaded image, but the image ends up corrupted each time the commit is made.

```
const readFile = (file: Blob) => {
  return new Promise((resolve, reject) => {
    const reader = new FileReader();

    reader.onload = (res) => {
      if (res.target) {
        resolve(res.target.result);
      }
    };
    reader.onerror = (err) => reject(err);

    reader.readAsBinaryString(file);
  });
};

const img = await readFile(images[0])
const buffer = Buffer.from(file, 'binary')

// I have also tried base64
const content = buffer.toString('utf8')

await github.createGitCommit({
  baseBranchName: BASE_BRANCH,
  headBranchName: branchName,
  commitMessage: `Added new event: ${state.title}`,
  files: [
    {
      path: `static/files/${images[0].name}`,
      content: content,
    }
  ],
});

```

Since [content must be in utf8](https://github.com/Himenon/github-api-create-commit/blob/main/src/types.ts#L23), I don't think committing images is currently supported. Every time I commit, the image ends up corrupted:

<img width="1511" alt="Screen Shot 2022-09-05 at 9 42 17 PM" src="https://user-images.githubusercontent.com/39626451/188548488-ed834ced-995c-487d-b020-35e5002e9c45.png">

**Comment by Himenon** [comment]
Good idea

**Comment by Himenon** [comment]
Released [v2.1.0](https://github.com/Himenon/github-api-create-commit/releases/tag/%40himenon%2Fgithub-api-create-commit%402.1.0)

I added encoding parameter

* https://github.com/Himenon/github-api-create-commit/blob/%40himenon/github-api-create-commit%402.1.0/src/types.ts#L21

**Comment by Himenon** [comment]
The issue is closed now that the operation has been verified.

**Comment by Himenon** [comment]
Sorry republished [v2.1.1](https://www.npmjs.com/package/@himenon/github-api-create-commit/v/2.1.2)

**Comment by AnimeAllstar** [comment]
I ended up achieving committing using the GitHub API: https://github.com/ubccsss/content-manager/blob/d4da4d8a4c173a4972e46ec0abac311fdf682fe7/src/utils/github.ts

I didn't end up using the package since I had to do a bunch of tasks besides committing.

It'd be nice if there were a package that abstracted a bunch of high-level commands like committing. The GitHub API is really low-level, and it was a pain to understand how it worked.

Thanks for the help.
